### PR TITLE
HPIS-109 | Edit business card editing

### PR DESCRIPTION
**Pull request by okrupa**

### CODE REVIEW

File: server/src/main/kotlin/pishello/hello/api/CardEndpoint.kt [hunk 1]
```diff
@@ -49,4 +50,23 @@ class CardEndpoint(val cardPort: CardPort, val photosStorage: PhotosStorage) {
...
     }
+
+    @PostMapping("/changedata", consumes = ["application/json"])
+    fun setCard(@RequestBody request: CardDataa): ResponseEntity<Unit> {
+        return if (cardPort.updateData(request.id, request.company, request.name, request.phone, request.email) != null) {
+            ResponseEntity(HttpStatus.OK)
+        } else {
+            ResponseEntity(HttpStatus.NOT_FOUND)
+        }
+//        print("Do dinasours still exist?\n")
```
Comment (BurnedOliveTree) [inline server/src/main/kotlin/pishello/hello/api/CardEndpoint.kt]: XD

File: server/src/main/kotlin/pishello/hello/api/CardEndpoint.kt [hunk 2]
```diff
@@ -49,4 +50,23 @@ class CardEndpoint(val cardPort: CardPort, val photosStorage: PhotosStorage) {
             ResponseEntity(HttpStatus.NOT_FOUND)
         }
     }
+
+    @PostMapping("/changedata", consumes = ["application/json"])
+    fun setCard(@RequestBody request: CardDataa): ResponseEntity<Unit> {
```
Comment (BurnedOliveTree) [inline server/src/main/kotlin/pishello/hello/api/CardEndpoint.kt]: so... there was already an endpoint to change data of the card: "/user/set" on line 29 in this file

File: server/src/main/kotlin/pishello/hello/api/SearchEndpoint.kt [hunk 3]
```diff
@@ -26,4 +28,41 @@ class SearchEndpoint(val searchPort: CardPort) {
         } else
             ResponseEntity(HttpStatus.NOT_FOUND)
     }
+
+    @PostMapping("/changedata", consumes = ["application/json"])
+    fun setCard(@RequestBody request: CardData): ResponseEntity<Unit> {
```
Comment (BurnedOliveTree) [inline server/src/main/kotlin/pishello/hello/api/SearchEndpoint.kt]: Why do we have a second endpoint that does nearly the same thing as /user/changedata ?

File: server/src/main/kotlin/pishello/hello/api/CardEndpoint.kt (hunk 1 above)
Comment (BurnedOliveTree) [reply server/src/main/kotlin/pishello/hello/api/CardEndpoint.kt]: [minor] you could delete all these comments

File: server/src/main/kotlin/pishello/hello/api/CardEndpoint.kt [hunk 4]
```diff
@@ -9,6 +9,7 @@ import pishello.hello.persistence.cloudStorage.PhotosStorage
 import pishello.hello.persistence.database.ports.CardPort
 
 data class CardRequest(val id: Int, val mode: String?, val category: String?)
+data class CardDataa(val id: Int, val company: String?, val name: String?, val phone: String?, val email: String?)
```
Comment (BurnedOliveTree) [inline server/src/main/kotlin/pishello/hello/api/CardEndpoint.kt]: [minor] Dataaaaaaa

File: frontend/actions/edit.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+export const editCardData = (idU, company,name, phone, email ) => {
+
+    return async dispatch => {
+        const id = parseInt(idU);
+        console.log('Updating data...')
+
+        
+        var response;
+
+        console.log("próbuje wysłać???????")
```
Comment (BurnedOliveTree) [inline frontend/actions/edit.js]: [minor] I would change it to English

File: server/src/main/kotlin/pishello/hello/persistence/database/ports/CardPort.kt [hunk 6]
```diff
@@ -43,4 +43,17 @@ class CardPort(val repository: CardRepository, val userRepository: UserRepositor
         card.path = path
         return repository.save(card)
     }
+
+    fun updateData(id: Int, company: String?, name: String?, phone: String?, email: String?): Card? {
```
Comment (BurnedOliveTree) [inline server/src/main/kotlin/pishello/hello/persistence/database/ports/CardPort.kt]: sadly, there is no way change mode or add category from application now, I would add it

**Review by BurnedOliveTree — CHANGES_REQUESTED** [review]
So, you added a separate endpoint for changing the data of cards, but that's kind of on me since I should't have added it in the first place. But there is no way to change card to public mode or to add a category to it, and that needs to change, since search won't work without it. I have tested it manually, and the whole workflow works, thought automatic tests would be also nice, but I won't block it if you don't add them.

File: server/src/main/kotlin/pishello/hello/api/SearchEndpoint.kt (hunk 3 above)
Comment (Rolkarolka) [reply server/src/main/kotlin/pishello/hello/api/SearchEndpoint.kt:33]: Bump

**Review by Rolkarolka — COMMENTED** [review]
Prośba również o dodanie do SQLki tworzącej tabele card nowe kolumny

**Review by BurnedOliveTree — DISMISSED** [review]

File: server/src/main/kotlin/pishello/hello/api/SearchEndpoint.kt (hunk 3 above)
Comment (Rolkarolka) [reply server/src/main/kotlin/pishello/hello/api/SearchEndpoint.kt]: Prośba o przeniesienie endpointu do klasy CardEndpoint

File: frontend/screens/MyBusinessCardsScreen.js [hunk 7]
```diff
@@ -59,7 +60,8 @@ const MyBusinessCardsScreen = props => {
             renderItem={({item}) => {
                 console.log(item);
                 return (
-                    <TouchableOpacity
+                    // onPress={() => navigation.navigate('Edit', {itemId: 86, otherParam: 'anything you want here',}
```
Comment (Rolkarolka) [inline frontend/screens/MyBusinessCardsScreen.js]: Prośba o usunięcie komentarza, skoro nie potrzebne

**Review by Rolkarolka — CHANGES_REQUESTED** [review]
Zostawiłam jeszcz 3 uwagi

**Review by BurnedOliveTree — COMMENTED** [review]
I would also move this endpoint to /card, it's more fitting there. Rest is good :D

**Review by Rolkarolka — COMMENTED** [review]
Prośba o dodanie testów :)

**Review by Rolkarolka — APPROVED** [review]

**Pull request merged**